### PR TITLE
feat: Added auto suggestions to various fields

### LINKED ISSUE #11: Add autocomplete and highlighting feature on headers, body and basic auth username text boxes
@Le-Roi777 added support for autocomplete and highlighting on target URL 🙏

image

But Headers, Body and Basic Auth Username text boxes do not contain autocomplete and highlighting features. 

image

Also, autocomplete must be case insensitive like [Ddosify Cloud](https://app.ddosify.com/). 

image

### PR DESCRIPTION
**Pull request by Lalit3716**
Fixes: #11

Added a reusable AutoSuggestionField component and used it in various places throughout the extension.

### CODE REVIEW

File: ui/src/App.js [hunk 1]
```diff
@@ -19,99 +19,99 @@ import {
...
-import CssBaseline from "@mui/material/CssBaseline";
-import React, { useEffect, useState } from "react";
-import "./App.css";
-import IconButton from "@mui/material/IconButton";
-import DeleteIcon from "@mui/icons-material/Delete";
-import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
-import MenuItem from "@mui/material/MenuItem";
-import jsPDF from "jspdf";
-import { dynamicVarsList } from "./dynamicVarsList";
+} from '@mui/material';
```
Comment (fatihbaltaci) [inline ui/src/App.js]: Hey @Lalit3716, I guess your code formatter changes all the double quotes into single quotes. I can not track the changes. Could you please change the single quotes into double quotes? Thanks.

Comment (Lalit3716) [reply ui/src/App.js]: Hi @fatihbaltaci, yes prettier indeed changed all double quotes into single quotes, but before that, the codebase was using inconsistent quotes all over the place.

I think consistent use of quotes is a little bit better but if you want to change all single quotes to double I can do that!

Comment (fatihbaltaci) [reply ui/src/App.js]: Please change all the single quotes into double quotes to solve the inconsistency. After that we will review the code, thanks a lot

Comment (Lalit3716) [reply ui/src/App.js]: Alright, I changed all single quotes to double quotes!

**Comment by fatihbaltaci** [comment]
Hi @Lalit3716, autocomplete and highlighting work great on headers and basic auth, thanks!

 Could you please add autocomplete and highlighting features on the **body** as well? Thanks.

**Comment by Lalit3716** [comment]
@fatihbaltaci Done!

**Comment by fatihbaltaci** [comment]
@Lalit3716 we found an issue with headers, could you please fix that? `master` branch does not have this issue. 


https://user-images.githubusercontent.com/10353650/198842851-885f0006-8d70-4a34-8f66-a8eab4a0da24.mov

**Comment by Lalit3716** [comment]
Hi @fatihbaltaci, thanks for letting me know this, I fixed the issue now everything should work as expected!

**Comment by fatihbaltaci** [comment]
Hi @Lalit3716, everything works as expected, thanks for your effort 🚀

**Pull request merged**